File: apps/mobile/src/types/brief.ts
export type BriefPriority =
  | 'low'
  | 'medium'
  | 'high';

export type Brief = {
  id: string;

  title: string;
  summary: string;

  key_points: string[];
  actions: string[];
  tags: string[];

  priority: BriefPriority;

  due_date?: string | null;
  created_at: string;
};

File: services/api/app/schemas/brief.py
from typing import List, Optional

from pydantic import BaseModel, Field


class BriefRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=10000)


class BriefResponse(BaseModel):
    title: str
    summary: str
    key_points: List[str] = Field(default_factory=list)
    actions: List[str] = Field(default_factory=list)
    tags: List[str] = Field(default_factory=list)
    priority: str
    due_date: Optional[str] = None